### PR TITLE
[AIRFLOW-5641] support running git sync container as root

### PR DESCRIPTION
**Pull request by houqp**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5641

### Description

Fixed a regression that prevents user from using root as user for git sync container.

Also changed `_get_security_context_val` so that it behaves as documented. 

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

See changed files in tests folder

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/kubernetes/worker_configuration.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def _get_init_containers(self) -> List[k8s.V1Container]:
 
         if self.kube_config.git_sync_run_as_user != "":
             init_containers.security_context = k8s.V1SecurityContext(
-                run_as_user=self.kube_config.git_sync_run_as_user or 65533
+                run_as_user=self.kube_config.git_sync_run_as_user
```
Comment (ashb) [inline airflow/kubernetes/worker_configuration.py:157]: Do we still apply this 65533 default UID somewhere?

**Review by ashb — CHANGES_REQUESTED** [review]
Test that if the git_sync_run_as_user is not specified at all that we still set the default of 65533 please

File: airflow/kubernetes/worker_configuration.py (hunk 1 above)
Comment (houqp) [reply airflow/kubernetes/worker_configuration.py:157]: Yes, the default comes from the default config, `self.kube_config.git_sync_run_as_user` will always contain a defined value if it's not `''`.

**Comment by houqp** [comment]
@ashb I tried testing that, Airflow exited on error because that config key exists in default config, so it is required. `('kubernetes', 'git_sync_run_as_user'): ''` is the closest thing to that test.

File: airflow/kubernetes/worker_configuration.py (hunk 1 above)
Comment (houqp) [reply airflow/kubernetes/worker_configuration.py:157]: this is now also covered by a test case.

**Comment by houqp** [comment]
Turns out I was setting gi_sync_run_as_user to None, which was causing the crash. I have added a test case to avoid setting this config key entirely, which results in the default 65533 uid.

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
This is only a problem against master right? (The code looks quite different on the release branch)

**Comment by ashb** [comment]
Marking this for inclusion in 1.10.11 -- but that depends on @dimberman's backporting work.

**Pull request merged**